src/database_lmdb.cpp

```diff
@@ -0,0 +1,346 @@
+/*
+Minetest
+Copyright (C) 2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "config.h"
+
+#if USE_LMDB
+
+#include <vector>
+#include <string>
+#include "irr_v3d.h"
+#include "irrlichttypes.h"
+#include "database.h"
+#include "database_lmdb.h"
+#include "settings.h"
+#include "exceptions.h"
+#include "lmdb.h"
+#include "filesys.h"
+#include <errno.h>
+
+#define MAKEKEY(pos)                    \
+	MDB_val key;                        \
+	s64 key_ = getBlockAsInteger(pos);  \
```

Comment:
> An alternative could be to use v3s16 directly as a key ?

Ehhh. No. I'm mistaken. It's a C++ class. It might even have virtual methods, and Compilers might append 2 bytes of padding or other housekeeping data, which would add 2 or more bytes of random data. Not a good idea.

An array of 3 s16 values (`s16 key[3]`) would be OK as a key.